src/wallet/wallet.h

```diff
@@ -505,22 +500,21 @@ class CWalletTx
             mapValueCopy["timesmart"] = strprintf("%u", nTimeSmart);
         }
 
-        std::vector<uint256> dummy_vector; //!< Used to be vMerkleBranch
-        s << tx << hashBlock << dummy_vector << nIndex;
-        std::vector<CMerkleTx> vUnused; //!< Used to be vtxPrev
-        s << vUnused << mapValueCopy << vOrderForm << fTimeReceivedIsTxTime << nTimeReceived << fFromMe << fSpent;
+        std::vector<char> dummy_vector1; //!< Used to be vMerkleBranch
+        std::vector<char> dummy_vector2; //!< Used to be vtxPrev
+        char dummy_char = false; //!< Used to be fSpent
+        s << tx << hashBlock << dummy_vector1 << nIndex << dummy_vector2 << mapValueCopy << vOrderForm << fTimeReceivedIsTxTime << nTimeReceived << fFromMe << dummy_char;
     }
 
     template<typename Stream>
     void Unserialize(Stream& s)
     {
         Init(nullptr);
-        char fSpent;
 
-        std::vector<uint256> dummy_vector; //!< Used to be vMerkleBranch
-        s >> tx >> hashBlock >> dummy_vector >> nIndex;
-        std::vector<CMerkleTx> vUnused; //!< Used to be vtxPrev
-        s >> vUnused >> mapValue >> vOrderForm >> fTimeReceivedIsTxTime >> nTimeReceived >> fFromMe >> fSpent;
+        std::vector<uint256> dummy_vector1; //!< Used to be vMerkleBranch
+        std::vector<CMerkleTx> dummy_vector2; //!< Used to be vtxPrev
```

Comment:
fad5decb1fdab249f626c6468174c50be15f14cf

Could remove this and inline `CMerkleTx::Unserialize`, see:

<details>

```diff
--- a/src/wallet/wallet.h
+++ b/src/wallet/wallet.h
@@ -365,26 +365,6 @@ struct COutputEntry
     int vout;
 };

-/** Legacy class used for deserializing vtxPrev for backwards compatibility.
- * vtxPrev was removed in commit 93a18a3650292afbb441a47d1fa1b94aeb0164e3,
- * but old wallet.dat files may still contain vtxPrev vectors of CMerkleTxs.
- * These need to get deserialized for field alignment when deserializing
- * a CWalletTx, but the deserialized values are discarded.**/
-class CMerkleTx
-{
-public:
-    template<typename Stream>
-    void Unserialize(Stream& s)
-    {
-        CTransactionRef tx;
-        uint256 hashBlock;
-        std::vector<uint256> vMerkleBranch;
-        int nIndex;
-
-        s >> tx >> hashBlock >> vMerkleBranch >> nIndex;
-    }
-};
-
 //Get the marginal bytes of spending the specified output
 int CalculateMaximumSignedInputSize(const CTxOut& txout, const CWallet* pwallet, bool use_max_sig = false);

@@ -512,9 +492,17 @@ public:
         Init(nullptr);

         std::vector<uint256> dummy_vector1; //!< Used to be vMerkleBranch
-        std::vector<CMerkleTx> dummy_vector2; //!< Used to be vtxPrev
         char dummy_char; //! Used to be fSpent
-        s >> tx >> hashBlock >> dummy_vector1 >> nIndex >> dummy_vector2 >> mapValue >> vOrderForm >> fTimeReceivedIsTxTime >> nTimeReceived >> fFromMe >> dummy_char;
+        s >> tx >> hashBlock >> dummy_vector1 >> nIndex;
+        {
+            CTransactionRef tx;
+            uint256 hashBlock;
+            std::vector<uint256> vMerkleBranch;
+            int nIndex;
+
+            s >> tx >> hashBlock >> vMerkleBranch >> nIndex; //!< Used to be vtxPrev
+        }
+        s >> mapValue >> vOrderForm >> fTimeReceivedIsTxTime >> nTimeReceived >> fFromMe >> dummy_char;

         ReadOrderPos(nOrderPos, mapValue);
         nTimeSmart = mapValue.count("timesmart") ? (unsigned int)atoi64(mapValue["timesmart"]) : 0;
```

</details>

And if you do so then I'd squash now that `CMerkleTx` is actually removed.

Reply:
I don't think that works, because the vector might have length 0, or more than 1

Reply:
This isn't equivalent. `vtxPrev` was a vector, so:

- there can be multiple `CMerkleTx` objects serialized
- the serialization is preceded with a compactSize

I would like to remove `CMerkleTx` entirely, but implementing the low-lever deserialization logic for a vector of `CMerkleTx`s here seems like it could be easy to introduce an inconsistency (potentially causing wallets to get corrupted) and it seems wrong to be adding low-level deserialization code to this file.

Reply:
Ah sorry.. You could read compact size, then loop to read dummy values but really not worth it. You can resolve this.

Reply:
> You could read compact size, then loop to read dummy values but really not worth it.

Exactly. Seems error-prone and redundant.